src/app/drive/components/ShareDialog/ShareDialog.tsx

```diff
@@ -355,7 +491,126 @@ const ShareDialog = (props) => {
+                                  <p className="w-full text-left text-base font-medium leading-none">
+                                    {translate('modals.shareModal.requests.actions.roles.viewer')}
+                                  </p>
+                                </button>
+
+                                {/* Editor */}
+                                <button
+                                  className="flex h-9 w-full cursor-pointer items-center justify-start space-x-3 rounded-lg px-3 hover:bg-gray-5"
+                                  onClick={() => {
+                                    onAcceptRequest(request.email, 'editor');
+                                    close();
+                                  }}
+                                >
+                                  <p className="w-full text-left text-base font-medium leading-none">
+                                    {translate('modals.shareModal.requests.actions.roles.editor')}
+                                  </p>
+                                </button>
+                              </>
+                            )}
+                          </Popover.Panel>
+                        </>
+                      )}
+                    </Popover>
+                    <Button variant="secondary" onClick={() => handleDenyRequest(request.email)}>
+                      <span>{translate('modals.shareModal.requests.actions.deny')}</span>
+                    </Button>
+                  </div>
+                </div>
+
+                {request.message ? (
+                  <div className="mb-3 flex flex-col space-y-1 rounded-lg bg-gray-5 p-4 text-base leading-tight">
+                    {request.message.split('\n').map((line) => (
+                      <p>{line}</p>
+                    ))}
+                  </div>
+                ) : (
+                  <></>
+                )}
+              </div>
+            ))
+          ) : (
+            <div className="absolute inset-0 flex items-center justify-center">
+              <div className="flex flex-col items-center justify-center space-y-1 rounded-2xl bg-gray-5 p-6 text-lg font-medium text-gray-50">
+                <CheckCircle weight="thin" size={64} />
+                <span>{translate('modals.shareModal.requests.empty')}</span>
+              </div>
+            </div>
+          )}
         </div>
+      ),
+    };
+
+    return view[viewProps.view];
+  };
+
+  return (
+    <Modal className="p-0" isOpen={isOpen} onClose={onClose} preventClosing={isLoading}>
+      {!isInvitingUser ? (
+        <>
+          <div className="flex h-16 w-full items-center justify-between space-x-4 border-b border-gray-10 px-5">
+            <Header view={view} />
+          </div>
+
+          <div className="flex flex-col space-y-4 p-5">
+            <View view={view} />
+          </div>
+        </>
       ) : (
         <ShareInviteDialog onInviteUser={onInviteUser} />
       )}
```

Comment:
@CandelR "invite" body should be imported in the <View /> component, not treated as an independent dialog.